content/en/docs/concepts/workloads/controllers/replicaset.md

```diff
@@ -88,66 +102,114 @@ Events:
+ReplicaSet, they will immediately be acquired by it.
+
+Creating the Pods after the frontend ReplicaSet has been deployed and set up its initial Pod replicas to fulfill its
+replica count requirement:
+
+```shell
+kubectl create -f http://k8s.io/examples/pods/pod-rs.yaml
+```
 
-The `.spec.template` is the only required field of the `.spec`. The `.spec.template` is a 
-[pod template](/docs/concepts/workloads/pods/pod-overview/#pod-templates). It has exactly the same schema as a 
-[pod](/docs/concepts/workloads/pods/pod/), except that it is nested and does not have an `apiVersion` or `kind`.
+Will cause the new Pods to be acquired by the ReplicaSet, then immediately terminated as the ReplicaSet would be over
+its desired count. Fetching the pods:
+```shell
+kubectl get pods
+```
 
-In addition to required fields of a pod, a pod template in a ReplicaSet must specify appropriate
-labels and an appropriate restart policy.
+Will show that the new Pods are either already terminated, or in the process of being terminated
+```shell
+NAME             READY   STATUS        RESTARTS   AGE
+frontend-9si5l   1/1     Running       0          1m
+frontend-dnjpy   1/1     Running       0          1m
+frontend-qhloh   1/1     Running       0          1m
+pod2             0/1     Terminating   0          4s
+```
 
-For labels, make sure to not overlap with other controllers. For more information, see [pod selector](#pod-selector).
+If we create the Pods first:
+```shell
+kubectl create -f http://k8s.io/examples/pods/pod-rs.yaml
+```
 
-For [restart policy](/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy), the only allowed value for `.spec.template.spec.restartPolicy` is `Always`, which is the default.
+And then create the ReplicaSet however:
+```shell
+kubectl create -f http://k8s.io/examples/controllers/frontend.yaml
+```
 
-For local container restarts, ReplicaSet delegates to an agent on the node,
-for example the [Kubelet](/docs/admin/kubelet/) or Docker.
+We shall see that the ReplicaSet has acquired the Pods and has only created new ones according to its spec until the
+number of its new Pods and the original matches its desired count. As fetching the pods:
+```shell
+kubectl get pods
+```
 
-### Pod Selector
+Will reveal in its output:
+```shell
+NAME             READY   STATUS    RESTARTS   AGE
+frontend-pxj4r   1/1     Running   0          5s
+pod1             1/1     Running   0          13s
+pod2             1/1     Running   0          13s
+```
 
-The `.spec.selector` field is a [label selector](/docs/concepts/overview/working-with-objects/labels/). A ReplicaSet
-manages all the pods with labels that match the selector. It does not distinguish
-between pods that it created or deleted and pods that another person or process created or
-deleted. This allows the ReplicaSet to be replaced without affecting the running pods.
+In this manner, a ReplicaSet can own a non-homogenous set of Pods
```

Comment:
At this point, reinforce the idea that you don't want to create Pods that match the selector of a current or future ReplicaSet. Be specific, instead of saying "non-homogeneous", I suggest saying something like, " a set of Pods that are running different images."